### DISCUSSION TITLE
File Upload with WSL2

### DISCUSSION THREAD

**Discussion by st11x** [thunderclient/thunder-client-support#806] [Q&A]
Hi,

I have seen this issue

https://github.com/rangav/thunder-client-support/issues/158

and tried to change the extension settings to 'ui' but it didn't work.

My extension is installed in WSL2, so for file upload, it should browse a WSL2 location but instead it opens up file explorer. This is unlike the export Collection feature where it opens the correct filesystem by default. 

![image](https://user-images.githubusercontent.com/7398499/192608359-c94ba916-3f57-427f-8469-ec79ca22f623.png)

Unlike issue 158, I am unable to edit the file location manually and it only shows the filename.

No matter where I choose, it appends the full location to the custom workspace location. I was hoping to specify a relative path like to the "files" folder in the thunder-client workspace so I can distribute the collection. How can I achieve that? 

thanks
  Matt

**Comment by rangav** [comment]
Hi Matt, remove that vscode setting and try again 

next try these settings below, see which one works

https://github.com/rangav/thunder-client-support/issues/158#issuecomment-850962620

**Reply by st11x** [reply]
@rangav I did that already. I mentioned the 'ui' setting.

I also experimented with combination of values. The file upload button only likes to operate in 'ui' mode, so I have to remove my custom workspace location setting which is in WSL2 otherwise the extension will fail to load.

The file upload button also generates paths with escaped backslashes even in 'workspace' (the default) mode. 

I find that I can get it to work only if I edit the file thunderclient.json directly and convert it to a path like:

"file/bulk.json"

from the path 

"\\\\wsl.localhost\\Ubuntu-22.04\\home\\msoh\\projects\\thunder-tests\\files\\bulk.json"

the extension writes that to the file and it end up getting concatenated with the custom workspace location. 

If I removed my custom workspace location, then the selected path doesn't get concatenated but then it also couldn't find the WSL location like so:

![image](https://user-images.githubusercontent.com/7398499/192618210-063213c4-fa65-4cae-a72e-cdc09c2bd1f1.png)




Matt

**Reply by rangav** [reply]
WSL issue is fixed now, please update to `v1.20.0` or newer

See all features released
https://github.com/rangav/thunder-client-support/releases/tag/v1.20.0

Please let me know your feedback.
